smartsim/settings/builders/launchArgBuilder.py

```diff
@@ -90,3 +95,8 @@ def format_env_vars(
     def __str__(self) -> str:  # pragma: no-cover
         string = f"\nLaunch Arguments:\n{fmt_dict(self._launch_args)}"
         return string
+
+
+class ExecutableLike(t.Protocol):
```

Comment:
is this just a holder? What is ExecutableLike for?

Reply:
I wasn't fully certain on what our API is going to look like, so I tried to keep my interface as vague as possible. See [this comment](https://github.com/CrayLabs/SmartSim/pull/620#discussion_r1664938340) for a more detailed explanation of `Protocol` specifically